### ISSUE TITLE
Setup.hs incompatible with new Cabal

### ISSUE THREAD

**Issue by ocharles** [bgamari/optimization#6]
```
unpacking source archive /nix/store/gi2va5wkcqglfcy30agqgrbpcabqcww0-optimization-0.1.7.tar.gz
source root is optimization-0.1.7
setting SOURCE_DATE_EPOCH to timestamp 1448533079 of file optimization-0.1.7/.vim.custom
patching sources
Run jailbreak-cabal to lift version restrictions on build inputs.
compileBuildDriverPhase
setupCompileFlags: -package-db=/tmp/nix-build-optimization-0.1.7.drv-0/package.conf.d -j2 -threaded
[1 of 1] Compiling Main             ( Setup.lhs, /tmp/nix-build-optimization-0.1.7.drv-0/Main.o )

Setup.lhs:8:31: error:
    Module
    ‘Distribution.Package’
    does not export
    ‘PackageName(PackageName)’
  |
8 | import Distribution.Package ( PackageName(PackageName), PackageId, InstalledPackageId, packageVersion, packageName )
  |                               ^^^^^^^^^^^^^^^^^^^^^^^^
builder for ‘/nix/store/v07lwbf9044qwk11q7sf3hbmar52rwwi-optimization-0.1.7.drv’ failed with exit code 1
```

**Comment by bgamari** [comment]
Thanks for bringing this to my attention!

**Comment by bgamari** [comment]
This was fixed with 7ea12dbe48244150100a25c299d2f080cd420c0f.
